Xamarin.PropertyEditing.Mac/Controls/CombinablePropertyEditor.cs

```diff
@@ -0,0 +1,137 @@
+		{
+			get { return (CombinablePropertyViewModel<T>)base.ViewModel; }
+			set { base.ViewModel = value; }
+		}
+
+		List<NSButton> combinableList = new List<NSButton> ();
+		NSView firstKeyView;
+		NSView lastKeyView;
+
+		protected override void HandleErrorsChanged (object sender, DataErrorsChangedEventArgs e)
+		{
+			UpdateErrorsDisplayed (ViewModel.GetErrors (e.PropertyName));
+		}
+
+		protected override void SetEnabled ()
+		{
+			foreach (var item in combinableList) {
+				item.Enabled = ViewModel.Property.CanWrite;
+			}
+		}
+
+		protected override void UpdateErrorsDisplayed (IEnumerable errors)
+		{
+			if (ViewModel.HasErrors) {
+				SetErrors (errors);
+			} else {
+				SetErrors (null);
+				SetEnabled ();
+			}
+		}
+
+		protected override void OnViewModelChanged (PropertyViewModel oldModel)
+		{
+			combinableList.Clear ();
+
+			const float controlHeight = 22;
+
+			// Set our new RowHeight
+			RowHeight = (ViewModel.Choices.Count * controlHeight);
+
+			float top = controlHeight;
+			foreach (var item in ViewModel.Choices) {
+				var BooleanEditor = new NSButton (new CGRect (4, RowHeight - top, Frame.Width - 4, controlHeight)) {
+					AllowsMixedState = true,
+					ControlSize = NSControlSize.Small,
+					Font = NSFont.FromFontName (DefaultFontName, DefaultFontSize),
+					Title = item.Name,
+					TranslatesAutoresizingMaskIntoConstraints = false,
+				};
+				BooleanEditor.SetButtonType (NSButtonType.Switch);
+				BooleanEditor.Activated += SelectionChanged;
+
+				AddSubview (BooleanEditor);
+				combinableList.Add (BooleanEditor);
+				top += controlHeight;
+			}
+
+			// Set our tabable order
+			firstKeyView = combinableList.First ();
+			lastKeyView = combinableList.Last ();
+
+			base.OnViewModelChanged (oldModel);
+		}
+
+		protected override void UpdateValue ()
+		{
+			foreach (var item in combinableList) {
+				if (ViewModel.Choices.Count > 0) {
+					var flag = ViewModel.Choices.FirstOrDefault (x => x.Name == item.Title);
+					if (flag.IsFlagged.HasValue) {
+						item.State = flag.IsFlagged.Value ? NSCellStateValue.On : NSCellStateValue.Off;
+						item.AllowsMixedState = false;
+					} else {
+						item.State = NSCellStateValue.Mixed;
+					}
+				} else {
+					item.State = NSCellStateValue.Off;
+				}
+			}
+		}
+
+		protected override void UpdateAccessibilityValues ()
+		{
+			foreach (var item in combinableList) {
+				item.AccessibilityEnabled = item.Enabled;
+				item.AccessibilityTitle = string.Format (LocalizationResources.AccessibilityCombobox, ViewModel.Property.Name);
+			}
+		}
+
+		void SelectionChanged (object sender, EventArgs e)
+		{
+			var button = sender as NSButton;
+
+			if (button != null) {
+				var choice = ViewModel.Choices.First (x => x.Name == button.Title);
+				switch (button.State) {
+					case NSCellStateValue.Off:
+						choice.IsFlagged = false;
+						break;
+					case NSCellStateValue.On:
+						choice.IsFlagged = true;
+						break;
+				}
```

Comment:
Q: in mixed state, which is not explicitly handled here, are we even getting here, and what should we be doing? Maybe add a comment.

Reply:
My understanding is that it can only ever be in a mixed state by either reset, or first time we load. So If a selection is made it can only be on or off. Am I making sense?

Reply:
Totally, that was kind of my assumption, but I wasn't sure. Thanks.

Reply:
> My understanding is that it can only ever be in a mixed state by either reset, or first time we load. 

Almost, another case is if a second object is selected that an individual flag is set for one but not the other.